lms/services/assignment.py

```diff
@@ -198,11 +200,19 @@ def get_by_id(self, id_: int) -> Assignment | None:
 
     def is_member(self, assignment: Assignment, user: User) -> bool:
         """Check if a user is a member of an assignment."""
-        return bool(
-            self._db.query(AssignmentMembership)
-            .filter_by(user=user, assignment=assignment)
-            .one_or_none()
-        )
+        return bool(assignment.membership.filter_by(user=user).one_or_none())
+
+    def get_members(
+        self, assignment, role_type: RoleType, role_scope: RoleScope
+    ) -> list[User]:
+        return [
+            member.user
+            for member in assignment.membership.options(
+                joinedload(AssignmentMembership.user)
```

Comment:
joinedload fetch all users in one go to avoid having to make a query per each one.